include/exadg/compressible_navier_stokes/spatial_discretization/kernels_and_operators.h

```diff
@@ -960,6 +960,15 @@ class ViscousOperator
   typedef dealii::Tensor<2, dim, dealii::VectorizedArray<Number>> tensor;
   typedef dealii::Point<dim, dealii::VectorizedArray<Number>>     point;
 
+  bool
+  using_simplex(FaceIntegratorScalar & integrator) const
+  {
+    return integrator.get_matrix_free()
+      .get_dof_handler()
+      .get_triangulation()
+      .all_reference_cells_are_simplex();
+  }
```

Comment:
I see this code multiple times, so we should not realize this design.

Reply:
Yes, I also didn't like it while I was copy-pasting the function. But I allowed myself doing it, since the Kernels in operators `diffusive_operator`, `viscous_operator` etc. have multiple identical functions and as far as I see no base class. 

As a short term solution, I can suggest pushing the function to `interior_penalty_parameter.h` as a free standing function.

Reply:
Which functions do you mean exactly? If this is the case, we should also aim to improve this, but we should not introduce new duplicated code. A free function makes sense in my opinion.

Reply:
Most of the Kernel functions are duplicated in different Kernels. Here are some random examples from `DiffusiveKernel` and `ViscousKernel`: 

**I will put the function in `interior_penalty_parameter.h`, but the best would be to create a base class for the Kernels, which is I think outside of the purpose of this PR.**

https://github.com/exadg/exadg/blob/3ea21b101c8b3a421a56fc706a38705e484a2837/include/exadg/incompressible_navier_stokes/spatial_discretization/operators/viscous_operator.h#L83-L88

https://github.com/exadg/exadg/blob/3ea21b101c8b3a421a56fc706a38705e484a2837/include/exadg/convection_diffusion/spatial_discretization/operators/diffusive_operator.h#L80-L85

https://github.com/exadg/exadg/blob/3ea21b101c8b3a421a56fc706a38705e484a2837/include/exadg/incompressible_navier_stokes/spatial_discretization/operators/viscous_operator.h#L120-L132

https://github.com/exadg/exadg/blob/3ea21b101c8b3a421a56fc706a38705e484a2837/include/exadg/convection_diffusion/spatial_discretization/operators/diffusive_operator.h#L87-L99

Reply:
I agree that such a refactoring of Kernels is not related to the present PR and should, therefore, not be done here.

Reply:
Moved the function to `grid.h`, as discussed.